specification/securityinsights/resource-manager/Microsoft.SecurityInsights/stable/2024-09-01/DataConnectors.json

```diff
@@ -0,0 +1,1696 @@
+          "type": "string",
+          "description": "API Key Identifier"
+        },
+        "isApiKeyInPostPayload": {
+          "type": "boolean",
+          "description": "Flag to indicate if API key is set in HTTP POST payload"
+        }
+      },
+      "required": [
+        "apiKey",
+        "apiKeyName"
+      ],
+      "x-ms-discriminator-value": "APIKey"
+    },
+    "AWSAuthModel": {
+      "description": "Model for API authentication with AWS.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/CcpAuthConfig"
+        }
+      ],
+      "type": "object",
+      "properties": {
+        "roleArn": {
+          "type": "string",
+          "description": "AWS STS assume role ARN",
+          "example": "arn:aws:iam::123456789012:role/your-role-name"
+        },
+        "externalId": {
+          "type": "string",
+          "description": "AWS STS assume role external ID. This is used to prevent the confused deputy problem: 'https://docs.aws.amazon.com/IAM/latest/UserGuide/confused-deputy.html'"
+        }
+      },
+      "required": [
+        "roleArn"
+      ],
+      "x-ms-discriminator-value": "AWS"
+    },
+    "BasicAuthModel": {
+      "description": "Model for API authentication with basic flow - user name + password.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/CcpAuthConfig"
+        }
+      ],
+      "type": "object",
+      "properties": {
+        "userName": {
+          "type": "string",
+          "description": "The user name."
+        },
+        "password": {
+          "type": "string",
+          "description": "The password"
+        }
+      },
+      "required": [
+        "userName",
+        "password"
+      ],
+      "x-ms-discriminator-value": "Basic"
+    },
+    "GCPAuthModel": {
+      "description": "Model for API authentication for all GCP kind connectors.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/CcpAuthConfig"
+        }
+      ],
+      "type": "object",
+      "properties": {
+        "serviceAccountEmail": {
+          "type": "string",
+          "description": "GCP Service Account Email"
+        },
+        "projectNumber": {
+          "type": "string",
+          "description": "GCP Project Number"
+        },
+        "workloadIdentityProviderId": {
+          "type": "string",
+          "description": "GCP Workload Identity Provider ID"
+        }
+      },
+      "required": [
+        "serviceAccountEmail",
+        "projectNumber",
+        "workloadIdentityProviderId"
+      ],
+      "x-ms-discriminator-value": "GCP"
+    },
+    "GenericBlobSbsAuthModel": {
+      "description": "Model for API authentication for working with service bus or storage account.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/CcpAuthConfig"
+        }
+      ],
+      "type": "object",
+      "properties": {
+        "credentialsConfig": {
+          "type": "object",
+          "additionalProperties": {
```

Comment:
>additionalProperties [](http://example.com/codeflow?start=11&length=20)

ARM is trying to reduce instances of unschematized APIs to make it easier for consumers of those APIs to understand them and use them reliably, not to mention provide better support in the various SDKs.

Please make this with a schema or provide an explanation for why you need this.